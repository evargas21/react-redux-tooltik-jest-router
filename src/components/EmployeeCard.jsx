// @packages
import * as ProtoTypes from 'prop-types';

const EmployeeCard = ({ contactNumber, document, fullName, id, onEdit }) => {
    const edit = () => onEdit(id);

    return (
        <div className="card">
            <div className="card-avatar">{fullName.charAt(0).toUpperCase()}</div>
            <div className="group-items">
                <div className="group-card">
                    <h3>{fullName}</h3>
                    <div className="list-button">
                        <button onClick={edit}>
                            <span className="material-icons">edit</span>
                        </button>
                    </div>
                </div>
                <div className="content-card">
                    <p><span className="material-icons">person</span>{document}</p>
                    <p><span className="material-icons">call</span>{contactNumber}</p>
                </div>
            </div>
        </div>
    );
};

EmployeeCard.propTypes = {
    contactNumber: ProtoTypes.string.isRequired,
    document: ProtoTypes.number.isRequired,
    fullName: ProtoTypes.string.isRequired,
    id: ProtoTypes.string.isRequired,
    onEdit: ProtoTypes.func.isRequired,
};

export default EmployeeCard;
