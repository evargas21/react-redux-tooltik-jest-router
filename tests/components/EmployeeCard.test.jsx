// @packages
import { render, screen } from '@testing-library/react';

// @scripts
import EmployeeCard from '../../src/components/EmployeeCard';

describe('Test in <EmployeeCard />',() => {

    const contact = '1234567';
    const document = 566667;
    const name = 'test name';
    const id = 'ee-11';

    test('must match snapshot', () => {
        const { container } = render(
            <EmployeeCard
                contactNumber={contact}
                document={document}
                fullName={name}
                id={id}
                onEdit={jest.fn()}
            />
        )
        expect(container).toMatchSnapshot();
    });

    test('must display the component with the indicated value', () => {
        render(
            <EmployeeCard
                contactNumber={contact}
                document={document}
                fullName={name}
                id={id}
                onEdit={jest.fn()}
            />
        )
        const title = screen.getByText(name.charAt(0).toUpperCase())
        expect(title).toBeTruthy();
    });
});
