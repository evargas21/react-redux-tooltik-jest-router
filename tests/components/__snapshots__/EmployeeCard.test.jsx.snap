// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <EmployeeCard /> must match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-avatar"
    >
      T
    </div>
    <div
      class="group-items"
    >
      <div
        class="group-card"
      >
        <h3>
          test name
        </h3>
        <div
          class="list-button"
        >
          <button>
            <span
              class="material-icons"
            >
              edit
            </span>
          </button>
        </div>
      </div>
      <div
        class="content-card"
      >
        <p>
          <span
            class="material-icons"
          >
            person
          </span>
          566667
        </p>
        <p>
          <span
            class="material-icons"
          >
            call
          </span>
          1234567
        </p>
      </div>
    </div>
  </div>
</div>
`;
